lib/utils.js

```diff
@@ -57,6 +57,12 @@ function stripHttp1ConnectionHeaders (headers) {
 // issue ref: https://github.com/fastify/fast-proxy/issues/42
 function buildURL (source, reqBase) {
   let baseOrigin = reqBase ? new URL(reqBase).href : undefined
+
+  // To make sure we don't accidentally override the base path
+  if (baseOrigin && source.startsWith('//')) {
```

Comment:
I think this could be faster.

```suggestion
  if (baseOrigin && source.length > 1 && source[0] === '/' && source[1] === '/') {
```

Reply:
we can get rid of length, no?

Reply:
changed, but I think it's less readable

Reply:
If source is just '/' then source[1] would be undefined and could result in a cache miss or some other deoptimization, which could have a bigger performance penalty than checking the length. 

But i am not sure.

@RafaelGSS 
Any opinion on this?

Reply:
updated